### PR TITLE
Cleanup System.Globalization.Extensions tests

### PR DESCRIPTION
**Pull request by hughbe**
- Combine tests to take advantage of xunit
- Renames tests to follow guidelines

Contributes to #2336

### CODE REVIEW

File: src/System.Globalization.Extensions/tests/IdnMapping/GetAsciiTests.cs [hunk 1]
```diff
@@ -1,154 +0,0 @@
...
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000", 0));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000", 0, 2));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000\u0101"));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000\u0101", 0));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000\u0101", 0, 3));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000\u0101\u0000"));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000\u0101\u0000", 0));
-            Assert.Throws<ArgumentException>(() => idn.GetAscii("\u0101\u0000\u0101\u0000", 0, 4));
-            Assert.Throws<ArgumentException>(() => idn.GetUnicode("abc\u0000", 0, 4));
-            Assert.Throws<ArgumentException>(() => idn.GetUnicode("ab\u0000c", 0, 4));
```
Comment (stephentoub) [inline src/System.Globalization.Extensions/tests/IdnMapping/GetAsciiTests.cs]: What happened to these last two tests?

Comment (stephentoub) [inline src/System.Globalization.Extensions/tests/IdnMapping/GetAsciiTests.cs]: Actually, I see the first one now, but the last one seems to have disappeared.

File: src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingThrows.cs [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-{
-    public class IdnMappingThrows
-    {
-        [Fact]
-        public static void GetAsciiThrows()
-        {
-            IdnMapping idnMapping = new IdnMapping();
-
-            Assert.Throws<ArgumentNullException>("unicode", () => idnMapping.GetAscii(null, -5));
-            Assert.Throws<ArgumentNullException>("unicode", () => idnMapping.GetAscii(null, -5, -10));
```
Comment (stephentoub) [inline src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingThrows.cs]: What happened to these tests that have both a null string and out-of-range values?

File: src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingThrows.cs [hunk 3]
```diff
@@ -1,40 +0,0 @@
...
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => idnMapping.GetAscii("abc", -5, -10));
```
Comment (stephentoub) [inline src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingThrows.cs]: What happened to this test that has both an out -of-range index and an out-of-range count?

**Comment by stephentoub** [comment]
A few tests that appear to have gone missing and should be added back, but other than that, LGTM.

**Comment by hughbe** [comment]
Thanks for the comments. I added back the tests and fixed up some other small stuff

**Comment by stephentoub** [comment]
@mmitche, mkdirs failure again:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/centos7.1_release_tst_prtest/2400/console

**Pull request merged**